### DISCUSSION TITLE
Hello Community, Filter between combos

### DISCUSSION THREAD

**Discussion by patriciomh1965** [serenity-is/Serenity#5320] [Help]
![combos](https://user-images.githubusercontent.com/42218266/103011514-d3cea900-4518-11eb-9b83-821fedb63602.png)
I have two combos with date data, the first one indicates yyyyMM, the second one indicates yyyyMMdd. When the user selects the yyyyMM of the first combo, the second combo should show only the yyyyMMdd dates.
I indicate the instructions and image:

XYZColumns
========

namespace CNC.Servicios.Columns
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using System;
    using System.ComponentModel;
    using System.Collections.Generic;
    using System.IO;
    using CNC.Modules.Servicios;

    [ColumnsScript("Servicios.TaControlProceso")]
    [BasedOnRow(typeof(Entities.TaControlProcesoRow), CheckNames = true)]
    public class TaControlProcesoColumns
    {
        [EditLink()]
        [Width(100)]
        public Int32 Numero { get; set; }
        [Width(100)]
        [LookupEditor(typeof(ComboPeriodoProcesoLooKup)), QuickFilter(CssClass = "hidden-xs")]   
        public Int32 Periodo { get; set; }
        [Width(100)]
        [LookupEditor(typeof(ComboControlLooKup)), QuickFilter(CssClass = "hidden-xs")]      
        //[QuickFilter(CssClass = "hidden-xs")]  
        [SortOrder(1, descending: true)]
        public Int32 Control { get; set; }
        [Width(100)]
        public Int32 Diario { get; set; }
        [Width(150)]
        public String GlosaWorkflow { get; set; } 
        [Width(100)]
        public Int32 HoraInicio { get; set; }
        [Width(100)]
        public Int32 HoraTermino { get; set; }
        [Width(100)]
        public Int32 Prioridad { get; set; }
        [Width(150)]
        public String NombreEstado { get; set; } 
        [Width(100)]
        public Int32 Total { get; set; }
        [Width(100)]
        public Int32 TotalAcepta { get; set; }
        [Width(100)]
        public Int32 TotalOmite { get; set; }
        [Width(100)]
        public Int32 TotalRechaza { get; set; }
        [Width(150)]
        public DateTime FechaInicio { get; set; }
        [Width(150)]
        public DateTime FechaCierre { get; set; }
        [Width(100)]
        public String NombreWork { get; set; }
    }
}

Lookup ( yyyyMM)
============
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace CNC.Modules.Servicios
{
    using CNC.Servicios.Entities;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using Serenity.Web;

    [LookupScript]
    public class ComboPeriodoProcesoLooKup : RowLookupScript<CNC.Servicios.Entities.TaControlProcesoRow> 
    {
        public ComboPeriodoProcesoLooKup() 
        {
            IdField = TextField = TaControlProcesoRow.Fields.Periodo.PropertyName;
        }
        protected override void PrepareQuery(SqlQuery query)
        {

            var fld = TaControlProcesoRow.Fields;
            query.Distinct(true)
                .Select(fld.Periodo)
                .Where(
                    new Criteria(fld.Periodo) != "" &
                    new Criteria(fld.Periodo).IsNotNull())
                .OrderBy(fld.Periodo);
        }

        protected override void ApplyOrder(SqlQuery query)
        {
        }
    }
}

Lookup  (yyyyMMdd)
==============
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace CNC.Modules.Servicios
{
    using CNC.Servicios.Entities;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using Serenity.Web;

    [LookupScript]
    public class ComboControlLooKup : RowLookupScript<CNC.Servicios.Entities.TaControlProcesoRow>
    {
        public ComboControlLooKup() 
        {
            IdField = TextField = TaControlProcesoRow.Fields.Control.PropertyName;
        }
        protected override void PrepareQuery(SqlQuery query)
        {

            var fld = TaControlProcesoRow.Fields;
            var fl2 = Convert.ToString(fld.Periodo);
            query.Distinct(true)
                .Select(fld.Control)
                .Where(
                    new Criteria(fld.Control) != "" &
                    new Criteria(fld.Control).IsNotNull())
                .OrderBy(fld.Control);
        }
        
        protected override void ApplyOrder(SqlQuery query)
        {
        }
    }
}

XYZRow
=====

namespace CNC.Servicios.Entities
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using Serenity.Data.Mapping;
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.IO;

    [ConnectionKey("Servicios"), Module("Servicios"), TableName("[dbo].[ta_control_proceso]")]
    [DisplayName("Monitor"), InstanceName("Monitor")]
    [ReadPermission("Administration:General")]
    [ModifyPermission("Administration:General")]
    public sealed class TaControlProcesoRow : Row, IIdRow
    {
        [DisplayName("Numero"), Column("prc_numero"), PrimaryKey]
        public Int32? Numero
        {
            get { return Fields.Numero[this]; }
            set { Fields.Numero[this] = value; }
        }

        [DisplayName("Periodo"), Column("prc_periodo"), NotNull]
        public Int32? Periodo
        {
            get { return Fields.Periodo[this]; }
            set { Fields.Periodo[this] = value; }
        }
       
        [DisplayName("Control"), Column("prc_control"), NotNull, QuickFilter]       
        public Int32? Control
        {
            get { return Fields.Control[this]; }
            set { Fields.Control[this] = value; }
        }

        [DisplayName("Diario"), Column("prc_diario"), NotNull]
        public Int32? Diario
        {
            get { return Fields.Diario[this]; }
            set { Fields.Diario[this] = value; }
        }        
        //    
        [DisplayName("Workflow"), Column("prc_workflow"), ForeignKey(typeof(TaConfigProcesoWorkflowRow)), LeftJoin("jWorkflow")]
        [LookupEditor(typeof(TaConfigProcesoWorkflowRow))]
        public Int32? Workflow
        {
            get { return Fields.Workflow[this]; }
            set { Fields.Workflow[this] = value; }
        }
        [DisplayName("Workflow")]
        [Origin("jWorkflow"), MinSelectLevel(SelectLevel.List)]
        public String GlosaWorkflow
        {
            get { return Fields.GlosaWorkflow[this]; }
            set { Fields.GlosaWorkflow[this] = value; }
        }
        //    
        [DisplayName("Hora"), Column("prc_hora_inicio"), NotNull]
        public Int32? HoraInicio
        {
            get { return Fields.HoraInicio[this]; }
            set { Fields.HoraInicio[this] = value; }
        }

        [DisplayName("Minuto"), Column("prc_hora_fin"), NotNull]
        public Int32? HoraTermino
        {
            get { return Fields.HoraTermino[this]; }
            set { Fields.HoraTermino[this] = value; }
        }

        [DisplayName("Prioridad"), Column("prc_prioridad"), NotNull]
        public Int32? Prioridad
        {
            get { return Fields.Prioridad[this]; }
            set { Fields.Prioridad[this] = value; }
        }
        //           
        [DisplayName("Estado"), Column("prc_estado"), ForeignKey("[dbo].[ta_param_proceso_estado]", "pest_numero"), LeftJoin("jEstado"), TextualField("NombreEstado")]
        [LookupEditor(typeof(TaParamProcesoEstadoRow))]
        public Int32? Estado
        {
            get { return Fields.Estado[this]; }
            set { Fields.Estado[this] = value; }
        }
        [DisplayName("Estado"), Expression("jEstado.[pest_glosa]")]
        public String NombreEstado
        {
            get { return Fields.NombreEstado[this]; }
            set { Fields.NombreEstado[this] = value; }
        }
        //   

        [DisplayName("Total"), Column("prc_total")]
        public Int32? Total
        {
            get { return Fields.Total[this]; }
            set { Fields.Total[this] = value; }
        }

        [DisplayName("Total Acepta"), Column("prc_total_acepta")]
        public Int32? TotalAcepta
        {
            get { return Fields.TotalAcepta[this]; }
            set { Fields.TotalAcepta[this] = value; }
        }

        [DisplayName("Total Omite"), Column("prc_total_omite")]
        public Int32? TotalOmite
        {
            get { return Fields.TotalOmite[this]; }
            set { Fields.TotalOmite[this] = value; }
        }

        [DisplayName("Total Rechaza"), Column("prc_total_rechaza")]
        public Int32? TotalRechaza
        {
            get { return Fields.TotalRechaza[this]; }
            set { Fields.TotalRechaza[this] = value; }
        }

        [DisplayName("Fecha Inicio"), Column("prc_fecha_inicio")]
        public DateTime? FechaInicio
        {
            get { return Fields.FechaInicio[this]; }
            set { Fields.FechaInicio[this] = value; }
        }

        [DisplayName("Fecha Cierre"), Column("prc_fecha_cierre")]
        public DateTime? FechaCierre
        {
            get { return Fields.FechaCierre[this]; }
            set { Fields.FechaCierre[this] = value; }
        }
        //           
        [DisplayName("Tipo Workflow"), Column("prc_tipo_workflow"), ForeignKey("[dbo].[ta_param_tipo_workflow]", "twf_numero"), LeftJoin("jWork"), TextualField("NombreWork")]
        [LookupEditor(typeof(TaParamTipoWorkflowRow))]
        public Int32? TipoWorkflow
        {
            get { return Fields.TipoWorkflow[this]; }
            set { Fields.TipoWorkflow[this] = value; }
        }
        [DisplayName("Tipo Workflow"), Expression("jWork.[twf_glosa]")]
        public String NombreWork
        {
            get { return Fields.NombreWork[this]; }
            set { Fields.NombreWork[this] = value; }
        }
        //   
        ////  VINCULACION 
        //[DisplayName(""), MasterDetailRelation(foreignKey: "log_proceso", MasterKeyField = "prc_numero"), NotMapped]
        //public List<TaControlProcesoLogRow> DetailList
        //{
        //    get { return Fields.DetailList[this]; }
        //    set { Fields.DetailList[this] = value; }
        //}

        IIdField IIdRow.IdField
        {
            get { return Fields.Numero; }
        }

        public static readonly RowFields Fields = new RowFields().Init();

        public TaControlProcesoRow()
            : base(Fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field Numero;
            public Int32Field Periodo;
            public Int32Field Control;
            public Int32Field Diario;
            public Int32Field Workflow;
            public Int32Field HoraInicio;
            public Int32Field HoraTermino;
            public Int32Field Prioridad;
            public Int32Field Estado;
            public Int32Field Total;
            public Int32Field TotalAcepta;
            public Int32Field TotalOmite;
            public Int32Field TotalRechaza;
            public DateTimeField FechaInicio;
            public DateTimeField FechaCierre;
            public Int32Field TipoWorkflow;
            //
            public StringField NombreEstado;
            public StringField NombreWork;
            public StringField GlosaWorkflow;            
            //////
            //public RowListField<TaControlProcesoLogRow> DetailList;
               
        }
    }
}

I appreciate any help. Atte.

**Comment by VictorTomaili** [comment]
Hey @patriciomh1965, we have discussion tab for question etc. I move this issue to there.
